source/shadow.c

```diff
@@ -297,6 +398,41 @@ static ShadowStatus_t validateName( const char * pString,
 
 /*-----------------------------------------------------------*/
 
+static ShadowStatus_t extractThingName( const char * pTopic,
+                                        uint16_t topicLength,
+                                        uint16_t * pConsumedTopicLength,
+                                        uint16_t * pThingNameLength )
+{
+    ShadowStatus_t shadowStatus;
+
+    /* Extract thing name. */
+    shadowStatus = validateName( &( pTopic[ *pConsumedTopicLength ] ),
+                                 topicLength - *pConsumedTopicLength,
+                                 pThingNameLength );
+
+    if( shadowStatus != SHADOW_SUCCESS )
+    {
+        shadowStatus = SHADOW_THINGNAME_PARSE_FAILED;
+        LogDebug( ( "Not related to Shadow. Failed to parse thing name in pTopic %s.", pTopic ) );
```

Comment:
This part of the error message is confusing - _"Not related to shadow."_.  What does that mean to a developer working with a source file named `shadow.c`?  Same comment at line 421, I'd simply delete this.

Reply:
This message was carried over from the old Shadow library. I think it is a good log message to keep as the API function `Shadow_MatchTopicString` attempt to parse the the topic provided to API and checks if it is a Shadow topic or not.  If there is a failure, I think it is a good idea to keep a log at debug level to indicate what does the topic miss from being a Shadow topic. However, I recognize your concern. I think updating the log to 
```
LogDebug( ( "Not a Shadow topic. Failed to parse thing name in pTopic %s.", pTopic ) );
```
What do you think about this?

Reply:
I guess that's okay but let's make sure we audit all of these and leave this modified text in only where it can be of help.  These also take up space so we don't want to be wasteful.